build_tools/hack/build_env_diag.py

```diff
@@ -0,0 +1,882 @@
+        sh $ source .venv/bin/activate
+    If you not generate it, we recommend you use 'uv' to create a virtual env and activate it:
+        PS > uv venv .venv --python <CPython_VERSION>
+        PS > .venv/Scripts/activate.ps1
+
+        sh $ uv venv .venv --python <CPython_VERSION>
+        sh $ source .venv/bin/activate
+
+    traceback: Detected {program} not using Virtual ENV or using Global ENV 
+    """, Color_Warning)
+
+else:
+    print(
+    f"{Emoji_Warning} [{program}] Cannot found {err(program)} program.")
+    printc(f"""
+    We cannot found {program} on your device.
+    TheRock and PyTorch have some python dependices will be installed by requirements.txt.
+    Create a Python with venv and activate it.
+    
+    traceback: No available {program} installed or {program} not in PATH
+    """, Color_Warning)
+
+## Chocolatey (Unnecessary).
+if WINDOWS:
+    program = "choco".lower()
+    path = where(program)
+    if path:
+        CHOCO_INSTALL_DIR, CHOCO_MAJOR_VERSION, CHOCO_MINOR_VERSION, CHOCO_PATCH_VERSION = where(program)
+        print(
+        f"{Emoji_Pass} [{program}] Found {program} {CHOCO_MAJOR_VERSION}.{CHOCO_MINOR_VERSION}.{CHOCO_PATCH_VERSION} at: {hint(CHOCO_INSTALL_DIR)}")
+    else:
+        print(
+        f"{Emoji_Warning} [{program}] Cannot found {warn(program)} program.")
+    printc(f"""
+    {Emoji_Warning} If you have already installed tools, or prefer manually install toolchain one-by-one with specified installers, you can ignore this message.       
+
+    We cannot find Chocolatey on your device.
+    Chocolatey is a package manager for Windows platform, almost required packages here can install via choco command.
+    You can install chocolatey via PowerShell command.
+    PS > Set-ExecutionPolicy Bypass -Scope Process -Force
+         [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072
+         iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1'))
+        
+    traceback: No available {program} installed or {program} not in PATH
+    """, Color_Warning)
+
+## Astral uv. 
+program = "uv".lower()
+path = where(program)
+if path:
+    UV_INSTALL_DIR, UV_MAJOR_VERSION, UV_MINOR_VERSION, UV_PATCH_VERSION = where(program)
+    print(
+    f"{Emoji_Pass} [{program}] Found {program} {UV_MAJOR_VERSION}.{UV_MINOR_VERSION}.{UV_PATCH_VERSION} at: {hint(UV_INSTALL_DIR)}")
+else:
+    print(
+    f"{Emoji_Error} [{program}] Cannot found {err(program)} program.")
+    printc(f"""
+    We need uv to manage/accelerate generate Python venv for building TheRock and its side projects(PyTorch).
+    On Windows platform: You can install it via your Global Env Python, chocolatey or Astral official powershell install script:
+      PS > powershell -ExecutionPolicy ByPass -c "irm https://astral.sh/uv/install.ps1 | iex"
+      PS > pip install uv
+      PS > choco install uv
```

Comment:
We haven't technically tested `uv` yet (known issue on https://github.com/ROCm/TheRock/issues/808), but fine to include this for the future. I might suggest making the language less forceful here.